### DISCUSSION TITLE
[Q&A] Math syntax improperly generated and/or parsed 80% of the time

### DISCUSSION THREAD

**Discussion by noahgsolomon** [lobehub/lobehub#1113] [General Question] [answered]
### 💻 Operating System

macOS

### 📦 Environment

Official Preview

### 🌐 Browser

Other

### 🐛 Bug Description

Over 80% of the time I have used lobe chat, which I exclusively use for math related topics, it generates math expressions and symbols with surrounding parentheses instead of whatever the proper delimiter which is in use with your markdown to html converter uses.

I had a similar problem for one of my projects, and my solution was to detect if it was a math problem, and if so, invoke the math function which simply gives the model deeper context as to how to type math. I used KaTex, and my delimiter was '$$'.

Here was my additional prompt:

```js
        `For mathematical expressions and code blocks, use appropriate markdown formatting. For example, for math we use RehypeKaTeX, and here are some math examples:

        - Simple expression: $$x$$ renders as x.
        - Power notation: $$x^2$$ renders as x^2.
        - Function with multiple variables: $$f(x, y, z)$$ renders as f(x, y, z).
        - Partial derivatives: $$\\frac{\\partial^2 f}{\\partial x \\partial y}$$ renders as \\frac{\\partial^2 f}{\\partial x \\partial y}.
        - Summation: $$\\sum_{i=1}^{n} i^2$$ renders as \\sum_{i=1}^{n} i^2.

        NOTE: In the examples above, I have used two backslashes where you see one. This is important because of the way strings resolve backslashes in JavaScript, requiring two backslashes to get one backslash in the string.`
```

### 🚦 Expected Behavior

It should parse the latex properly to render the proper math html.

### 📷 Recurrence Steps

![image](https://github.com/lobehub/lobe-chat/assets/111200060/8fa4793c-c9a9-495d-a6e6-a9b2f961a2bc)
Just ask it to explain equations and most times it will generate improperly

**Comment by noahgsolomon** [comment]
I can submit a PR if you guys would like an example implementation. It might be challenging to write this additional enforcement in each request, so maybe it would be helpful to use double parentheses (which the OpenAI models often print maybe because this is the delimiter it was trained on for ChatGPT) as an additional delimiter for math.

**Comment by arvinxx** [comment, marked as answer]
maybe you can set a system role prompt to make openai output correct syntax ?

**Comment by noahgsolomon** [comment]
![Screenshot 2024-01-20 at 11 55 11 AM](https://github.com/lobehub/lobe-chat/assets/111200060/da08c52d-f456-49a4-abac-82b93e3d0fc0)
Good idea, that worked 👍

**Reply by agjs** [reply]
Can you please share your system prompt? 

Thanks a lot
